src/Aspire.Hosting/Exec/ExecResourceManager.cs

```diff
@@ -0,0 +1,213 @@
+            type = "running";
+        }, cancellationToken);
+
+        // in the background wait for the exec resource to reach terminal state. Once done we can complete logging
+        _ = Task.Run(async () =>
+        {
+            await _resourceNotificationService.WaitForResourceAsync(execResource.Name, targetStates: KnownResourceStates.TerminalStates, cancellationToken).ConfigureAwait(false);
+            _resourceLoggerService.Complete(dcpExecResourceName); // complete stops the `WatchAsync` async-foreach below
+        }, cancellationToken);
+
+        await foreach (var logs in _resourceLoggerService.WatchAsync(dcpExecResourceName).WithCancellation(cancellationToken).ConfigureAwait(false))
+        {
+            foreach (var log in logs)
+            {
+                yield return new CommandOutput
+                {
+                    Text = log.Content,
+                    IsErrorMessage = log.IsErrorMessage,
+                    LineNumber = log.LineNumber,
+                    Type = type
+                };
+            }
+        }
+    }
+
+    public IResource? ConfigureExecResource()
+    {
+        if (!_execOptions.Enabled)
+        {
+            return null;
+        }
+
+        var targetResource = _model.Resources.FirstOrDefault(x => x.Name.Equals(_execOptions.ResourceName, StringComparisons.ResourceName));
+        if (targetResource is not IResourceSupportsExec targetExecResource)
+        {
+            _logger.LogWarning("Target resource '{ResourceName}' does not support exec.", _execOptions.ResourceName);
+            throw new ArgumentException($"Target resource '{_execOptions.ResourceName}' does not support exec");
+        }
+
+        var execResource = BuildResource(targetExecResource);
+
+        _logger.LogInformation("Resource '{ResourceName}' has been successfully built and added to the model resources.", execResource.Name);
+        _execResourceInitialized.SetResult(execResource);
+        return execResource;
+    }
+
+    IResource BuildResource(IResourceSupportsExec targetExecResource)
+    {
+        return targetExecResource switch
+        {
+            ProjectResource prj => BuildAgainstProjectResource(prj),
+            _ => throw new NotImplementedException(nameof(targetExecResource))
+        };
+    }
+
+    private IResource BuildAgainstProjectResource(ProjectResource project)
+    {
+        var projectMetadata = project.GetProjectMetadata();
+        var projectDir = Path.GetDirectoryName(projectMetadata.ProjectPath) ?? throw new InvalidOperationException("Project path is invalid.");
+        var (exe, args) = ParseCommand();
+
+        // unique name for the new exec resource
+        string execResourceName;
+        do
+        {
+            var shortId = Guid.NewGuid().ToString("N").Substring(0, 8);
+            execResourceName = "exec" + shortId;
+        } while (_model.Resources.Any(x => x.Name.Equals(execResourceName, StringComparisons.ResourceName)));
```

Comment:
Do we really need this?

Reply:
its for uniquesness of the new resource. if user has resource "exec" then there will be duplicate names - not sure if aspire will run correctly in that case, so decided to do some random name here.

Reply:
Add a suffix -exec to the resource you're a surrogate for